From 4998209d1f9b9557d6669a8f73e60e1b750835f3 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Thu, 28 Dec 2017 11:21:12 +0800
Subject: [PATCH 0481/1240] drm/i915/gvt: avoid unncessary reset in GVT-g guest

When i915 boots up, it will trigger a reset, but it's unnecessary
in GVT-g environment, so remove this reset which can reduce guest
boot time.

Change-Id: Id8c120c3229118af3c41fb1ef4ddbfbf71cb69fe
Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/i915_gem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 90359c7..e0a0f22 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -4532,7 +4532,7 @@ void i915_gem_sanitize(struct drm_i915_private *i915)
 	 * it may impact the display and we are uncertain about the stability
 	 * of the reset, so this could be applied to even earlier gen.
 	 */
-	if (INTEL_GEN(i915) >= 5) {
+	if (INTEL_GEN(i915) >= 5 && !intel_vgpu_active(i915)) {
 		int reset = intel_gpu_reset(i915, ALL_ENGINES);
 		WARN_ON(reset && reset != -ENODEV);
 	}
-- 
2.7.4

